lib/matplotlib/path.py

```diff
@@ -136,21 +136,26 @@ def __init__(self, vertices, codes=None, _interpolation_steps=1,
         else:
             vertices = np.asarray(vertices, np.float_)
 
+        if (vertices.ndim != 2) or (vertices.shape[1] != 2):
+            msg = "'vertices' must be a 2D list or array with shape Nx2"
+            raise ValueError(msg)
+
         if codes is not None:
             codes = np.asarray(codes, self.code_type)
-            assert codes.ndim == 1
-            assert len(codes) == len(vertices)
-            if len(codes):
-                assert codes[0] == self.MOVETO
+            if (codes.ndim != 1) or len(codes) != len(vertices):
+                msg = "'codes' must be a 1D list or array with the same length"
+                msg += "of 'vertices'"
+                raise ValueError(msg)
+            if len(codes) and codes[0] != self.MOVETO:
+                msg = "The first element of 'code' must be equal to 'MOVETO':"
+                msg += " {}"
+                raise ValueError(msg.format(self.MOVETO))
         elif closed:
             codes = np.empty(len(vertices), dtype=self.code_type)
             codes[0] = self.MOVETO
             codes[1:-1] = self.LINETO
             codes[-1] = self.CLOSEPOLY
 
-        assert vertices.ndim == 2
```

Comment:
has been moved a [few lines above](https://github.com/montefra/matplotlib/blob/no_assert/lib/matplotlib/path.py#L139). It makes more sense to check first `vertices`, that is a mandatory argument, and then compare with an optional one